### PR TITLE
RFR: Add X-Request-ID API headers so we can trace request in logs easily.

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
Example:

```
(virtualenv)/m/s/s/st2 git:support_trace_id_in_cli_run ❯❯❯ grep "a64fd291-f9cf-4797-89e8-b4a58c8f12db" logs/st2api.log
2015-08-19 16:42:38,875 140597257636112 INFO log [-] a64fd291-f9cf-4797-89e8-b4a58c8f12db -  GET /v1/rules with filters={} (remote_addr='127.0.0.1',method='GET',filters={},request_id='a64fd291-f9cf-4797-89e8-b4a58c8f12db',path='/v1/rules')
(virtualenv)/m/s/s/st2 git:support_trace_id_in_cli_run ❯❯❯
```

```
(virtualenv)/m/s/s/st2 git:support_trace_id_in_cli_run ❯❯❯ http http://localhost:9101/v1/rules             ✭ ✱ ◼
HTTP/1.1 200 OK
Access-Control-Allow-Headers: Content-Type,Authorization,X-Auth-Token,X-Request-ID
Access-Control-Allow-Methods: GET,POST,PUT,DELETE,OPTIONS
Access-Control-Allow-Origin: http://127.0.0.1:9101/
Access-Control-Expose-Headers: Content-Type,X-Limit,X-Total-Count,X-Request-ID
Connection: keep-alive
Content-Length: 30806
Content-Type: application/json; charset=UTF-8
Date: Wed, 19 Aug 2015 23:42:38 GMT
X-Request-Id: a64fd291-f9cf-4797-89e8-b4a58c8f12db
X-Total-Count: 38
```

```
(virtualenv)/m/s/s/st2 git:support_trace_id_in_cli_run ❯❯❯ http http://localhost:9101/v1/rules                         ✭ ✱ ◼
HTTP/1.1 500 Internal Server Error
Access-Control-Allow-Headers: Content-Type,Authorization,X-Auth-Token,X-Request-ID
Access-Control-Allow-Methods: GET,POST,PUT,DELETE,OPTIONS
Access-Control-Allow-Origin: http://127.0.0.1:9101/
Access-Control-Expose-Headers: Content-Type,X-Limit,X-Total-Count,X-Request-ID
Connection: keep-alive
Content-Length: 36
Content-Type: application/json
Date: Wed, 19 Aug 2015 23:50:24 GMT
X-Request-Id: 15612f18-21b7-47b9-82fb-34657b7be2d5

{
    "faultstring": "500 testing"
}

(virtualenv)/m/s/s/st2 git:support_trace_id_in_cli_run ❯❯❯

(virtualenv)/m/s/s/st2 git:support_trace_id_in_cli_run ❯❯❯ grep "15612f18-21b7-47b9-82fb-34657b7be2d5" logs/st2api.log
2015-08-19 16:50:24,186 140260114315536 INFO log [-] 15612f18-21b7-47b9-82fb-34657b7be2d5 -  GET /v1/rules with filters={} (remote_addr='127.0.0.1',method='GET',filters={},request_id='15612f18-21b7-47b9-82fb-34657b7be2d5',path='/v1/rules')
(virtualenv)/m/s/s/st2 git:support_trace_id_in_cli_run ❯❯❯
```

### CODE REVIEW

File: st2common/st2common/hooks.py [hunk 1]
```diff
@@ -273,3 +282,28 @@ def after(self, state):
             log_msg = '%(method)s %(path)s' % values
```
Comment (manasdk) [inline st2common/st2common/hooks.py]: Can you include the request_id in the after section also.

**Comment by manasdk** [comment]
Works for me.

+1

**Comment by Kami** [comment]
Ideally we would pass this ID through the whole pipeline, but that's a start.

**Comment by lakshmi-kannan** [comment]
Scrutinizer internal error I suppose. Merging to master.

**Comment by manasdk** [comment]
FYI - https://tools.ietf.org/html/rfc6648 

Looks like `X-` prefixes are not recommended any longer.

**Pull request merged**